src/python/pants/backend/go/build_integration_test.py

```diff
@@ -35,116 +36,85 @@ def rule_runner() -> RuleRunner:
-    "main.go",
-    textwrap.dedent(
-        """\
-    package main
-
-    import (
-    \t"fmt"
-    )
-
-    func main() {
-    \tfmt.Println("Hello world!")
-    }
-    """
-    ).encode("utf-8"),
-)
-
-LIB_SOURCE = FileContent(
-    "lib.go",
-    textwrap.dedent(
-        """\
-    package lib
-
-    import (
-    \t"fmt"
-    )
+def build_package(rule_runner: RuleRunner, binary_target: Target) -> BuiltPackage:
+    field_set = GoBinaryFieldSet.create(binary_target)
+    return rule_runner.request(BuiltPackage, [field_set])
 
-    func Quote(s string) string {
-    \treturn fmt.Sprintf(">> %s <<", s)
-    }
-    """
-    ).encode("utf-8"),
-)
-
-MAIN_USING_LIB_SOURCE = FileContent(
-    "main.go",
-    textwrap.dedent(
-        """\
-    package main
-
-    import (
-    \t"fmt"
-    \t"example.com/lib"
-    )
 
-    func main() {
-    \tfmt.Println(lib.Quote("Hello world!"))
-    }
-    """
-    ).encode("utf-8"),
-)
-
-
-def make_target(
-    rule_runner: RuleRunner,
-    source_files: List[FileContent],
-    *,
-    import_path: str,
-    dependencies: Iterable[Address] = (),
-    target_name: str = "target",
-) -> Target:
-    for source_file in source_files:
-        rule_runner.create_file(f"{source_file.path}", source_file.content.decode())
-    source_files_str = ", ".join(f'"{sf.path}"' for sf in source_files)
-    deps_str = ", ".join(f'"{addr.spec}"' for addr in dependencies)
-    rule_runner.add_to_build_file(
-        "",
-        f"go_package(name='{target_name}', import_path='{import_path}', sources=[{source_files_str}], dependencies=[{deps_str}])\n",
-    )
-    return rule_runner.get_target(Address("", target_name=target_name))
-
-
-def build_package(
-    rule_runner: RuleRunner,
-    main_target: Target,
-) -> BuiltPackage:
-    args = ["--backend-packages=pants.backend.go"]
-    rule_runner.set_options(args)
-    rule_runner.add_to_build_file(
-        "", f"go_binary(name='bin', binary_name='foo', main='{main_target.address.spec}')\n"
-    )
-    go_binary_target = rule_runner.get_target(Address("", target_name="bin"))
-    built_package = rule_runner.request(BuiltPackage, (GoBinaryFieldSet.create(go_binary_target),))
-    return built_package
-
-
-def test_package_one_target(rule_runner: RuleRunner) -> None:
+def test_package_simple(rule_runner: RuleRunner) -> None:
     rule_runner.write_files(
-        {"BUILD": "go_module(name='root')\n", "go.mod": "module foo.example.com\n"}
+        {
+            "go.mod": "module foo.example.com\n",
+            "BUILD": "go_module(name='root')\n",
+            "project/main.go": dedent(
+                """\
+                package main
+
+                import (
+                    "fmt"
+                )
+
+                func main() {
+                    fmt.Println("Hello world!")
+                }
+                """
+            ),
+            "project/BUILD": dedent(
+                """\
+                go_package(name='main', import_path='main')
```

Comment:
The `import_path='main'` is unnecessary. The rules will automatically use `main` as the package name when compiling a binary's `main` package.

Reply:
Also the name can be omitted and just use the default name.